HBN/dictionary.py

```diff
@@ -0,0 +1,198 @@
+                            equals = i_split.split('=')
+
+                            # if there's two entries it's "vanilla"
+                            if len(equals) == 2:
+                                coding = str(equals[0].strip())
+                                meaning = str(equals[1].strip())
+                                levels_dic[coding] = meaning
+
+                    # if there's only one entry after splitting on semi-colons, but splitting that one entry on the equal sign is length=2...
+                    elif len(splits.split('=')) == 2:
+                        coding, meaning = splits.split('=')
+                        if coding != '' and meaning != '':
+                            levels_dic[str(coding).strip()] = str(meaning).strip()
+                        else:
+                            print(ShortName, '| len(semicolons) == 1:', splits)
+
+                    # otherwise something's wrong so just print the debugging info
+                    else:
+                        print(basename, '|', ShortName, '|')
+
+                # if the Notes are empty, then just set Levels to None
+                else:
+                    levels_dic = None
+
+            # if an exception happens, then just set Levels to None
+            except:
+                levels_dic = None
+
+            # try to get the DataType
+            try:
+                DataType = str(data['Variable Type'][i])
+            except:
+                DataType = None
+
+            # Merge columns from Response value on Diagnosis_KSADS
+            if basename == 'Diagnosis_KSADS':
+                # Merge the separate entries into one column
+                data['Values'] = data[['Response Values', 'Unnamed: 3', 'Unnamed: 4', 'Unnamed: 5', 'Unnamed: 6', 'Unnamed: 7']].apply(
+                    lambda x: '; '.join(x[x.notnull()]), axis=1)
+
+            # try to get the ValueRange range of possibilities
+            try:
+                ValueRange = str(data['Values'][i])
+            except:
+                try:
+                    ValueRange = str(data['Value'][i])
+                except:
+                    ValueRange = None
+
+            # set new lines
+            ValueRange = ValueRange.replace('\n', '; ')
+            if ValueRange == 'nan':
+                ValueRange = None
+
+            # build the dictionary in the ShortName field
+            dictionary[ShortName] = {}
+            if Description:
+                dictionary[ShortName]["Description"] = Description
+                del Description
+            if DataType:
+                dictionary[ShortName]["DataType"] = DataType
+                del DataType
+            if ValueRange:
+                dictionary[ShortName]["ValueRange"] = ValueRange
+                del ValueRange
+            if levels_dic:
+                dictionary[ShortName]["Levels"] = levels_dic
+                del levels_dic
```

Comment:
Please put the `DataType` and `ValueRange` after the `Levels` if you're going to keep those BIDS-invalid fields. It's mainly just my preference to put the BIDS-invalid fields after the BIDS-valid fields so it's more obvious they don't belong.

```suggestion
            if levels_dic:
                dictionary[ShortName]["Levels"] = levels_dic
                del levels_dic
            if DataType:
                dictionary[ShortName]["DataType"] = DataType
                del DataType
            if ValueRange:
                dictionary[ShortName]["ValueRange"] = ValueRange
                del ValueRange
```